tokio/src/sync/mpsc/bounded.rs

```diff
@@ -22,6 +22,50 @@ pub struct Sender<T> {
     chan: chan::Tx<T, Semaphore>,
 }
 
+/// A Sender that does not influence RAII semantics, i.e. if all [`Sender`]
+/// instances of a channel were dropped and only `WeakSender` instances remain,
+/// the channel is closed.
+///
+/// In order to send messages, the `WeakSender` needs to be upgraded using
+/// [`WeakSender::upgrade`], which returns `Option<Sender>`, `None` if all
```

Comment:
This sentence doesn't really make sense at this comma. How about this?

> which returns `Option<Sender>`. It returns `None` if all `Sender`s have been dropped, and otherwise it returns a `Sender`.